### PR TITLE
Decompiles z_bg_jya_goroiwa

### PR DESCRIPTION
**Pull request by i82orbom**
* BgJyaGoroiwa_Init
	* func_808979C0
	* BgJyaGoroiwa_Destroy
	* func_80897970
	* func_80897A2C
	* BgJyaGoroiwa_Draw
	* func_80897DF0
	* func_80897DDC
	* BgJyaGoroiwa_Update
	* func_80897B48

### CODE REVIEW

File: src/overlays/actors/ovl_Bg_Jya_Goroiwa/z_bg_jya_goroiwa.c [hunk 1]
```diff
@@ -21,25 +33,181 @@ const ActorInit Bg_Jya_Goroiwa_InitVars = {
...
+
+static InitChainEntry initChain[] = {
+    ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneForward, 1000, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneScale, 500, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneDownward, 1000, ICHAIN_STOP),
+};
+
+void func_80897970(BgJyaGoroiwa* this) {
+    Sphere16* worldSphere = &this->collider.list->dim.worldSphere;
```
Comment (ethteck) [inline src/overlays/actors/ovl_Bg_Jya_Goroiwa/z_bg_jya_goroiwa.c:61]: Our current style is to have all declarations separated from the rest of the function via an empty line

File: src/overlays/actors/ovl_Bg_Jya_Goroiwa/z_bg_jya_goroiwa.c [hunk 2]
```diff
@@ -21,25 +33,181 @@ const ActorInit Bg_Jya_Goroiwa_InitVars = {
...
+    Collider_SetJntSph(ctx, colliderJntSph, &this->actor, &sJntSphInit, this->colliderItems);
+    func_80897970(this);
+    this->collider.list->dim.worldSphere.radius = 0x3A;
+}
+
+void func_80897A2C(BgJyaGoroiwa* this) {
+    Actor* thisx = &this->actor;
+
+    f32 rotFactor = 175.30046f;
+    f32 posDiff = thisx->posRot.pos.x - thisx->pos4.x;
```
Comment (ethteck) [inline src/overlays/actors/ovl_Bg_Jya_Goroiwa/z_bg_jya_goroiwa.c:80]: Normally we'd put all the declarations together, followed by an empty line. I get why you spaced this the way you did, though. Idk.

File: src/overlays/actors/ovl_Bg_Jya_Goroiwa/z_bg_jya_goroiwa.c [hunk 3]
```diff
@@ -21,25 +33,181 @@ const ActorInit Bg_Jya_Goroiwa_InitVars = {
...
+    this->unk_1B0 = 1.0f;
+}
+
+void func_80897B48(BgJyaGoroiwa* this, GlobalContext* globalCtx) {
+    Actor* thisx = &this->actor;
+    f32 tmpf1;
+    f32 tmpf2;
+    s16 tmp16;
+
+    tmpf1 = (-100.0f - thisx->posRot.pos.y) * 2.5f;
```
Comment (ethteck) [inline src/overlays/actors/ovl_Bg_Jya_Goroiwa/z_bg_jya_goroiwa.c:117]: You should be able to initialize this variable on the same line it's declared

File: src/overlays/actors/ovl_Bg_Jya_Goroiwa/z_bg_jya_goroiwa.h [hunk 4]
```diff
@@ -6,9 +6,17 @@
...
 struct BgJyaGoroiwa;
 
+typedef void (*BgJyaGoroiwaFunc)(struct BgJyaGoroiwa*, GlobalContext*);
+
 typedef struct BgJyaGoroiwa {
     /* 0x0000 */ Actor actor;
-    /* 0x014C */ char unk_14C[0x70];
+    /* 0x014C */ BgJyaGoroiwaFunc actionFunc;
+    /* 0x0150 */ ColliderJntSph collider;
+    /* 0x0170 */ ColliderJntSphItem colliderItems[1];
```
Comment (ethteck) [inline src/overlays/actors/ovl_Bg_Jya_Goroiwa/z_bg_jya_goroiwa.h:15]: Does this need to be an array?

File: src/overlays/actors/ovl_Bg_Jya_Goroiwa/z_bg_jya_goroiwa.c [hunk 5]
```diff
@@ -1,3 +1,9 @@
+/*
+ *  File: z_bg_jya_goroiwa.c
+ *  Overlay: ovl_Bg_Jya_Goroiwa
+ *  Description: Rolling Boulder
+ *  moves very slowly in some cases
+ */
 #include "z_bg_jya_goroiwa.h"
```
Comment (Roman971) [inline src/overlays/actors/ovl_Bg_Jya_Goroiwa/z_bg_jya_goroiwa.c:7]:
```suggestion
 */

#include "z_bg_jya_goroiwa.h"
```

File: src/overlays/actors/ovl_Bg_Jya_Goroiwa/z_bg_jya_goroiwa.c [hunk 6]
```diff
@@ -8,8 +14,14 @@ void BgJyaGoroiwa_Init(Actor* thisx, GlobalContext* globalCtx);
 void BgJyaGoroiwa_Destroy(Actor* thisx, GlobalContext* globalCtx);
 void BgJyaGoroiwa_Update(Actor* thisx, GlobalContext* globalCtx);
 void BgJyaGoroiwa_Draw(Actor* thisx, GlobalContext* globalCtx);
+void func_80897DF0(BgJyaGoroiwa* this, GlobalContext* globalCtx);
+void func_80897B48(BgJyaGoroiwa* this, GlobalContext* globalCtx);
+void func_808979C0(BgJyaGoroiwa* this, GlobalContext* ctx);
```
Comment (Roman971) [inline src/overlays/actors/ovl_Bg_Jya_Goroiwa/z_bg_jya_goroiwa.c:19]: We generally don't forward declare functions unless necessary, which isn't the case for some of these functions.
```suggestion
```

File: src/overlays/actors/ovl_Bg_Jya_Goroiwa/z_bg_jya_goroiwa.c [hunk 7]
```diff
@@ -8,8 +14,14 @@ void BgJyaGoroiwa_Init(Actor* thisx, GlobalContext* globalCtx);
 void BgJyaGoroiwa_Destroy(Actor* thisx, GlobalContext* globalCtx);
 void BgJyaGoroiwa_Update(Actor* thisx, GlobalContext* globalCtx);
 void BgJyaGoroiwa_Draw(Actor* thisx, GlobalContext* globalCtx);
+void func_80897DF0(BgJyaGoroiwa* this, GlobalContext* globalCtx);
```
Comment (Roman971) [inline src/overlays/actors/ovl_Bg_Jya_Goroiwa/z_bg_jya_goroiwa.c:17]: Also it's slightly better to separate generic actor functions from internal ones.
```suggestion
void BgJyaGoroiwa_Draw(Actor* thisx, GlobalContext* globalCtx);

void func_80897DF0(BgJyaGoroiwa* this, GlobalContext* globalCtx);
```

File: src/overlays/actors/ovl_Bg_Jya_Goroiwa/z_bg_jya_goroiwa.c [hunk 8]
```diff
@@ -21,25 +33,181 @@ const ActorInit Bg_Jya_Goroiwa_InitVars = {
...
+
+static ColliderJntSphInit sJntSphInit = {
+    { COLTYPE_UNK10, 0x11, 0x00, 0x39, 0x20, COLSHAPE_JNTSPH },
+    1,
+    sJntSphItemsInit,
+};
+
+static CollisionCheckInfoInit collisionCheckInfoInit = { 0x01, 0xF, 0x0, 0xFE };
+
+static InitChainEntry initChain[] = {
```
Comment (Roman971) [inline src/overlays/actors/ovl_Bg_Jya_Goroiwa/z_bg_jya_goroiwa.c:53]: We generally call CollisionCheckInfoInit variables `sColChkInfoInit`.
Also, in our naming scheme, all "static" variables (in the sense of local to the file) should start with `s`, while global variables start with `g`
```suggestion
static CollisionCheckInfoInit sColChkInfoInit = { 0x01, 0xF, 0x0, 0xFE };

static InitChainEntry sInitChain[] = {
```

File: src/overlays/actors/ovl_Bg_Jya_Goroiwa/z_bg_jya_goroiwa.c [hunk 9]
```diff
@@ -21,25 +33,181 @@ const ActorInit Bg_Jya_Goroiwa_InitVars = {
...
+    if (this->unk_1B6 >= 0x3D) {
+        func_80897B1C(this);
+        this->unk_1B0 = 0.1f;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Goroiwa/func_80897DDC.s")
+void BgJyaGoroiwa_Update(Actor* thisx, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+    f32 padding;
```
Comment (Roman971) [inline src/overlays/actors/ovl_Bg_Jya_Goroiwa/z_bg_jya_goroiwa.c:189]:
```suggestion
    s32 pad;
```

File: src/overlays/actors/ovl_Bg_Jya_Goroiwa/z_bg_jya_goroiwa.c [hunk 10]
```diff
@@ -21,25 +33,181 @@ const ActorInit Bg_Jya_Goroiwa_InitVars = {
...
+
+void func_80897970(BgJyaGoroiwa* this) {
+    Sphere16* worldSphere = &this->collider.list->dim.worldSphere;
+
+    worldSphere->center.x = this->actor.posRot.pos.x;
+    worldSphere->center.y = (s32)(this->actor.posRot.pos.y + 59.5f);
+    worldSphere->center.z = this->actor.posRot.pos.z;
+}
+
+void func_808979C0(BgJyaGoroiwa* this, GlobalContext* ctx) {
```
Comment (Roman971) [inline src/overlays/actors/ovl_Bg_Jya_Goroiwa/z_bg_jya_goroiwa.c:68]:
```suggestion
void func_808979C0(BgJyaGoroiwa* this, GlobalContext* globalCtx) {
```

File: src/overlays/actors/ovl_Bg_Jya_Goroiwa/z_bg_jya_goroiwa.c [hunk 11]
```diff
@@ -21,25 +33,181 @@ const ActorInit Bg_Jya_Goroiwa_InitVars = {
...
+    ColliderJntSph* colliderJntSph = &this->collider;
```
Comment (Roman971) [inline src/overlays/actors/ovl_Bg_Jya_Goroiwa/z_bg_jya_goroiwa.c:69]: This collider temp is probably not needed here.

Comment (i82orbom) [reply src/overlays/actors/ovl_Bg_Jya_Goroiwa/z_bg_jya_goroiwa.c:69]: Wasn't needed, but then a padding is needed to match the function...

Comment (Roman971) [reply src/overlays/actors/ovl_Bg_Jya_Goroiwa/z_bg_jya_goroiwa.c:71]: Padding seems better in this case

Comment (i82orbom) [reply src/overlays/actors/ovl_Bg_Jya_Goroiwa/z_bg_jya_goroiwa.c]: I agree

**Review by Roman971 — APPROVED** [review]

**Pull request merged**